site/source/tutorials/10XX_development_environment/index.md

```diff
@@ -0,0 +1,146 @@
+
+# Development Environment
+
+## Overview
+
+This tutorial will cover how to set up your tools and environment for the best development experience. You can use demo code from any of the tutorials, but this tutorial will reference finished code from the [Forms tutorial](../005_form_widgets/).
+
+## Prerequisites
+You can [download](../assets/10xx_development_environment-initial.zip) a fresh install of the demo code to get started.
+
+You should be familiar with the developer tools in your browser of choice. This tutorial will reference the Chrome developer tools, but the instructions should apply to any browser.
+
+{% section %}
+
+## Command line tools
+
+The [first tutorial](../000_local_installation/) goes into installing the Dojo CLI, and the available commands. Before that point, we assume you have an environment with supported versions of node, npm, and git installed.
+
+While git is not strictly necessary to use Dojo 2, a robust development environment also needs some sort of version control system. All of Dojo 2 uses git as the version control tool and we manage all of our code on GitHub. Several of READMEs and other instructions will assume that git is being used for version control. If it or another version control tool is not already installed, you can follow [these instructions](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).
+
+{% task 'Initialize a git repository' %}
+
+{% instruction 'Run `git init` within the root `biz-e-corp` directory' %}
+
+This should log `Initialized empty Git repository in /path/to/biz-e-corp/.git/` in your console.
+
+{% instruction 'Create your first commit' %}
+
+Use `git add` to begin tracking files, and `git commit` to create a commit. For more information, refer to [these instructions](https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository).
+
+{% section %}
+
+## Configure your editor
+
+{% task 'Get an editor that supports Typescript.' %}
+
+Any code editor will allow you to work on a Dojo 2 project, but an editor that supports TypeScript will give you a richer development experience, as Dojo 2 is built with TypeScript and specifically designed to take advantage of its features. A list of editors with plugins that enable typescript support is [available here](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Editor-Support).
+
+{% instruction 'Install Typescript support' %}
+
+In Atom, for example, this means installing [atom-typescript](https://atom.io/packages/atom-typescript). Visual Studio Code comes with typescript support by default, and for Vim try installing a combination of plugins for [language features](https://github.com/Quramy/tsuquyomi) and [syntax highlighting](https://github.com/leafgarland/typescript-vim).
+
+{% instruction 'Test error highlighting' %}
+
+Within the demo app, open `src/widgets/WorkerContainer.ts`. In it, there should be code to create an instance of the Worker widget on line 21. Add a fake property, e.g.:
```

Comment:
line 21 ... there's risk in referencing a specific line number here in case that tutorial changes... not sure if there's a better way to handle this risk however.